src/landIce/evaluators/LandIce_BasalFrictionCoefficient_Def.hpp

```diff
@@ -533,8 +644,65 @@ operator() (const BasalFrictionCoefficient_Tag&, const int& cell) const {
         }
 
       }
-    }
-  }
+      if (beta_type == BETA_TYPE::DEBRIS_FRICTION) {
+        ParamScalarT bedRoughnessValue;
+	ParamScalarT bulkFrictionValue;
+	ParamScalarT basalDebrisValue;
+        switch (bedRoughness_type) {
+        case FIELD_TYPE::FIELD:
+          bedRoughnessValue = bedRoughnessField(cell,ipt);
+          break;
+        case FIELD_TYPE::EXPONENT_OF_FIELD_AT_NODES:
+          if(nodal)
+            bedRoughnessValue = std::exp(bedRoughnessField(cell,ipt));
+          else {
+            bedRoughnessValue = 0;
+            for (int node=0; node<numNodes; ++node)
+              bedRoughnessValue += std::exp(bedRoughnessField(cell,node))*BF(cell,node,ipt);
+          }
+          break;
+        case FIELD_TYPE::EXPONENT_OF_FIELD:
+          bedRoughnessValue = std::exp(bedRoughnessField(cell,ipt));
+          break;
+        case FIELD_TYPE::CONSTANT:
+          bedRoughnessValue = bedRoughness;
+          break;
+        }
+
+	switch (bulkFriction_type) {
+	case FIELD_TYPE::FIELD:
+	  bulkFrictionValue = bulkFrictionField(cell,ipt);
+	  break;
+	case FIELD_TYPE::CONSTANT:
+	  bulkFrictionValue = bulkFriction;
+	  break;
+	}
+
+	switch (basalDebris_type) {
+	case FIELD_TYPE::FIELD:
+	  basalDebrisValue = basalDebrisField(cell,ipt);
+	  break;
+	case FIELD_TYPE::CONSTANT:
+	  basalDebrisValue = basalDebris;
+	  break;
+	} 
+
+        const double secsInYr = 365*24*3600;
+        const double scaling = secsInYr*pow(1000,n+1); //turns flow rate from [Pa^{-n} s^{-1}] to [k^{-1} kPa^{-n} yr^{-1}]
+        switch (flowRate_type) {
+        case FLOW_RATE_TYPE::CONSTANT: { 
+          beta(cell,ipt) /=  std::pow ( u_norm(cell,ipt) + bedRoughnessValue*scaling*flowRate_val*std::pow(NVal,n),  power); //bedRoughness in km
+//          beta(cell,ipt) = ((muValue * NVal * std::pow (u_norm(cell,ipt), power-1.0)) / std::pow ( u_norm(cell,ipt) + bedRoughnessValue*scaling*flowRate_val*std::pow(NVal,n), power)) + (bulkFrictionValue * NVal / u_norm(cell,ipt)) + (basalDebrisValue * u_norm(cell,ipt));
```

Comment:
Were you thinking to uncomment this? As it is, the basal debris field and the bulk friction field are unused...

Reply:
yes, this should be uncommented thanks for catching that!